Rasa_Bot/actions/helper.py

```diff
@@ -1226,23 +1226,65 @@ def get_faik_text(user_id):
     return kit_text, filled, activity_ids_list
 
 
-def get_daily_step_goal_from_db() -> int:
+def get_daily_step_goal_from_db(user_id) -> int:
     """
-    Get daily step goal for a given user from the database.
+    Get daily step goal for a given user using data from the step count database.
+
+    For example, for a single participant, daily step count over the last 9 days (ranked from
+    lowest to highest) was 1250, 1332, 3136, 5431, 5552, 5890, 6402, 7301, 10,103. In this case,
+    the 60th percentile represents a goal of 5890 steps. The 6th element is used.
+
+    If there is not data for 9 days available, the number of days will be supplemented to 9 by
+    adding the average of the days with data.
 
     Args:
     user_id (int): The user ID for whom the daily step goal is to be retrieved from the database.
 
     Returns:
     int: The daily step goal for the given user, retrieved from the database.
     """
+    steps_per_day = []
+
+    end = datetime.now()
+    start = end - timedelta(days=9)
+    steps_data = get_steps_data(user_id=user_id, start_date=start, end_date=end)
+
+    for day in steps_data:
+        steps_per_day.append(day['steps'])
+
+    if len(steps_per_day) < 9:
+        while len(steps_per_day) < 9:
+            steps_per_day.append(np.mean(steps_per_day))  # Add with the avg value up to 9 values
```

Comment:
```suggestion
    for _ in range(9 - len(steps_per_day)):
        steps_per_day.append(np.mean(steps_per_day))  # Add with the avg value up to 9 values
```

It's always better to avoid the while cycles.
Also, the mean will be updated every time you update the list, is that fine?

Reply:
I commited the suggestion. The mean will not change if the added sample is equal to the mean.